Automatically generated `.nuxt/tsconfig.json` excludes `modules` folder.

### Environment

-

### Reproduction

https://stackblitz.com/edit/github-jhzonc

### Describe the bug

Creating the following local module

```
.nuxt
modules/
└── hello/
    ├── runtime/
    │   └── plugin.ts
    └── index.ts
tsconfig.json
```

will autogenerate the following `.nuxt/tsconfig.json`:

```json
"include": [
    "../modules/hello/runtime",
  ],
  "exclude": [
    "../modules/hello",
    "../modules/hello/runtime/server",
  ]
```

So both `modules/hello/index.ts` and `modules/hello/runtime/plugin.ts` files do not inherit the tsconfig. My code (with errors as comments):

`index.ts`
```ts
import { addPlugin, createResolver, defineNuxtModule } from '@nuxt/kit'

export default defineNuxtModule({
  setup() {
    console.log('Hi from module!')
    
    // ts-error: The 'import.meta' meta-property is only allowed when the '--module'...
    const { resolve } = createResolver(import.meta.url) 

    addPlugin({
      src: resolve('./runtime/plugin'),
    })
  },
})
```

`plugin.ts`
```ts
// ts-error: Cannot find module '#imports' or its corresponding type declarations
import { defineNuxtPlugin } from '#imports'

export default defineNuxtPlugin(() => {
  console.log('Hi from plugin')
})
```

Removing `"../modules/hello"` from the `exclude` array will then remove the ts-errors, as expected.

So, is this expected behaviour? Is there a way to disable it? Am I missing something?

Any help will be really appreciated. 

Thanks in advanced.

P.S.

I am also using Volar Takeover mode, which made me realize that I do not even understand why are those errors thrown in the first place if both files are explicitly excluded. But I guess that's a separate volar question/issue. 


And my root `tsconfig.json` just extends the autogenerated config as recommended:

```json
{
  "extends": "./.nuxt/tsconfig.json"
}
```






### Additional context

_No response_

### Logs

_No response_

--- a/packages/kit/src/module/install.ts
+++ b/packages/kit/src/module/install.ts
@@ -42,7 +42,7 @@ export async function installModule (moduleToInstall: string | NuxtModule, inlin
 
 // --- Internal ---
 
-function getDirectory (p: string) {
+export function getDirectory (p: string) {
   try {
     // we need to target directories instead of module file paths themselves
     // /home/user/project/node_modules/module/index.js -> /home/user/project/node_modules/module

--- a/packages/kit/src/template.ts
+++ b/packages/kit/src/template.ts
@@ -8,6 +8,7 @@ import type { TSConfig } from 'pkg-types'
 import { readPackageJSON } from 'pkg-types'
 
 import { tryResolveModule } from './internal/esm'
+import { getDirectory } from './module/install'
 import { tryUseNuxt, useNuxt } from './context'
 import { getModulePaths } from './internal/cjs'
 import { resolveNuxtModule } from './resolve'
@@ -110,14 +111,14 @@ export async function updateTemplates (options?: { filter?: (template: ResolvedN
   return await tryUseNuxt()?.hooks.callHook('builder:generateApp', options)
 }
 export async function writeTypes (nuxt: Nuxt) {
-  const modulePaths = getModulePaths(nuxt.options.modulesDir)
+  const nodeModulePaths = getModulePaths(nuxt.options.modulesDir)
 
   const rootDirWithSlash = withTrailingSlash(nuxt.options.rootDir)
 
-  const modules = await resolveNuxtModule(rootDirWithSlash,
+  const modulePaths = await resolveNuxtModule(rootDirWithSlash,
     nuxt.options._installedModules
       .filter(m => m.entryPath)
-      .map(m => m.entryPath)
+      .map(m => getDirectory(m.entryPath))
   )
 
   const tsConfig: TSConfig = defu(nuxt.options.typescript?.tsConfig, {
@@ -150,11 +151,11 @@ export async function writeTypes (nuxt: Nuxt) {
         .filter(srcOrCwd => !srcOrCwd.startsWith(rootDirWithSlash) || srcOrCwd.includes('node_modules'))
         .map(srcOrCwd => join(relative(nuxt.options.buildDir, srcOrCwd), '**/*')),
       ...nuxt.options.typescript.includeWorkspace && nuxt.options.workspaceDir !== nuxt.options.rootDir ? [join(relative(nuxt.options.buildDir, nuxt.options.workspaceDir), '**/*')] : [],
-      ...modules.map(m => join(relativeWithDot(nuxt.options.buildDir, m), 'runtime'))
+      ...modulePaths.map(m => join(relativeWithDot(nuxt.options.buildDir, m), 'runtime'))
     ],
     exclude: [
       ...nuxt.options.modulesDir.map(m => relativeWithDot(nuxt.options.buildDir, m)),
-      ...modules.map(m => join(relativeWithDot(nuxt.options.buildDir, m), 'runtime/server')),
+      ...modulePaths.map(m => join(relativeWithDot(nuxt.options.buildDir, m), 'runtime/server')),
       // nitro generate output: https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/core/nitro.ts#L186
       relativeWithDot(nuxt.options.buildDir, resolve(nuxt.options.rootDir, 'dist'))
     ]
@@ -215,7 +216,7 @@ export async function writeTypes (nuxt: Nuxt) {
     ...nuxt.options._modules
   ]
     .filter(f => typeof f === 'string')
-    .map(async id => ({ types: (await readPackageJSON(id, { url: modulePaths }).catch(() => null))?.name || id })))
+    .map(async id => ({ types: (await readPackageJSON(id, { url: nodeModulePaths }).catch(() => null))?.name || id })))
 
   if (nuxt.options.experimental?.reactivityTransform) {
     references.push({ types: 'vue/macros-global' })

